src/Pdp/PublicSuffixListManager.php

```diff
@@ -69,18 +70,28 @@ public function refreshPublicSuffixList()
         $publicSuffixListArray = $this->parseListToArray(
             $this->cacheDir . '/' . self::PDP_PSL_TEXT_FILE
         );
-        $this->writePhpCache($publicSuffixListArray);
+
+        // do not empty existing PHP cache file if source TXT is empty
+        if (is_array($publicSuffixListArray) && !empty($publicSuffixListArray)) {
```

Comment:
I think we don't need throw an exception in this case, because we just leave the PHP cache file untouched.